addons/stock/models/product.py

```diff
@@ -189,7 +189,7 @@ def _compute_quantities_dict(self, lot_id, owner_id, package_id, from_date=False
                     0.0,
                 )
                 continue
-            rounding = product.uom_id.rounding
+            rounding = product._origin.uom_id.rounding
```

Comment:
you should add comment to the code explaining why it must be `_origin`  (and what it actually means).

Also, what if we change uom from one value to another value with different rounding. Will it use the new value, or the old one?

Reply:
Changes are done!!

If we change uom from one value to another value with different rounding it use the old value.

Reply:
We should we use old value and not the new one?

Reply:
https://github.com/odoo/odoo/blob/50c0d39b54d2b4ac51ae43d33b826ba730429c0b/addons/stock/models/product.py#L192

Here we change the UOM in product.template but this compute method is in product.product and for rounding they use the product variant so we can receive the old value when once you can save the record then you can receive the new value.

Reply:
It doesn't make sense to me. You are saying that after saving it will use the new value. Why should we use old value during onchange request?